From 6d2e7c3ea80cc1ca9c0399dbf16ec190c5a1f604 Mon Sep 17 00:00:00 2001
Message-Id: <6d2e7c3ea80cc1ca9c0399dbf16ec190c5a1f604.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 3 Dec 2013 11:30:52 +0530
Subject: [PATCH 115/312] FOR_UPSTREAM [VPG]: drm/i915: Bring UP Power Wells
 before disabling RC6

During the runtime power management we faced some random issue around
RC6 during the suspend. After some analyzing, we found by following the
flow of doing Forcewake Get before the RC6 Disable and forcewake Put
after we enable the RC6 solved most of the issues. This patch adds the
Forcewake get before disabling the RC6.

Issue: VIZ-3393
Change-Id: Id2055cabbe2c9a6fd4d1090f8e3a0d87de5f2e7b
Signed-off-by: Deepak S <deepak.s@intel.com>
Reviewed-on: https://android.intel.com/164317
Tested-by: sys_abtbuild <sys_abtbuild@intel.com>
Reviewed-by: Bloomfield, Jon <jon.bloomfield@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 67d5253..0df13fc 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3739,9 +3739,15 @@ static void valleyview_disable_rps(struct drm_device *dev)
 {
 	struct drm_i915_private *dev_priv = dev->dev_private;
 
+	/* we're doing forcewake before Disabling RC6,
+	 * This what the BIOS expects when going into suspend */
+	gen6_gt_force_wake_get(dev_priv, FORCEWAKE_ALL);
+
 	/* Disable rc6 */
 	vlv_set_rc6_mode(dev, true);
 
+	gen6_gt_force_wake_put(dev_priv, FORCEWAKE_ALL);
+
 	/* Disable rps */
 	vlv_set_rps_mode(dev, true);
 
-- 
1.7.9.5

